core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -59,9 +59,19 @@ Kokkos::View<uint32_t *, HIPSpace> hip_global_unique_token_locks(
 }  // namespace Kokkos
 
 namespace Kokkos {
-
 namespace Impl {
 
+namespace {
+
+using ScratchGrain = Kokkos::HIP::size_type[Impl::HIPTraits::WarpSize];
+constexpr auto sizeScratchGrain = sizeof(ScratchGrain);
+
+std::size_t scratch_count(const std::size_t size) {
+  return (size + sizeScratchGrain - 1) / sizeScratchGrain;
```

Comment:
I am not convinced defining that function is a good change but I will leave it up to @Rombur

Reply:
@rombur approved, and its what CUDA did so there is some consistency.